cfgov/v1/blocks.py

```diff
@@ -276,6 +277,44 @@ class NavGroup(blocks.StructBlock):
         label='Menu items')
 
 
+class NavFooter(blocks.StructBlock):
+    draft = blocks.BooleanBlock(
+        required=False,
+        default=False,
+        help_text='If checked, this block will only show '
+        'on our sharing site (Content).',
+        label='Mark block as draft'
+    )
+    content = blocks.RichTextBlock(
+        required=False,
+        features=['link']
+    )
+
+    def clean(self, value):
```

Comment:
If I follow this logic, this is taking what the user entered into the CMS editor, parsing it for links, modifying those links by adding attributes, and then saving that modified content in the database?

Is it possible instead to alter the links at serve time, like we already do extensively in something like `v1.parse_links`, or through a Wagtail editor hook of some kind? The downside of altering at save time is that we're persisting this particular implementation of link markup into the database - if in future it ever changes, we'll have to reprocess every link.

This code also makes some assumptions about the internal Wagtail logic around how links are stored in the DB (`expand_db_html` and `RichText` are undocumented). This makes it less future-proof than modifying at serve time.

Reply:
@chosak That's a good point. I originally handled the parsing at serve time, but thought it might be cleaner to perform it once on save instead. If that's not the case because of possible future incompatibilities, I can switch back to parsing the links when the menu items are assembled.

Reply:
@chosak @virginiacc I'd actually be a fan of moving away from parsing for links at serve time (something I've been looking into as part of my work on how we can best manage external links) since it's redundant and also requires that people remember to put in the `parse_links` jinja filter when editing or creating a new template...  I plan to do a more complete write-up on why the current implementation doesn't work/ is buggy) but I hear the concerns about using `expand_db_html`.

Reply:
@richaagarwal sure, I'd be interested to hear your thoughts.

Generally my main argument against storing these kinds of modifications in the DB is that it violates separation of content and presentation. The editor should be concerned with content only: what is being linked to and what should the link text say, and not need to care/think about HTML attributes. Things like link icons (`<a class>`) or labels (`<a aria-label>`) are part of the presentation, not the content. We don't want to have to think about modifying the page content every time we change the page presentation.

Reply:
@chosak Right, so I think there's a distinction between doing heavy processing to determine whether something is an external link, and then the actual mark-up that gets applied (which I agree would ideally happen at serve time).  But right now we're doing both at serve time, and we only should ever need to do the first part once / at page save time.  I know this particular PR isn't concerned with determining whether something is an external link or not, but if we do move away from having the `parse_links` filter called everywhere (inconsistently), then that wouldn't be as viable of an alternative.

Reply:
@chosak @richaagarwal I switched to handling the links at serve time in https://github.com/cfpb/cfgov-refresh/pull/4267/commits/3c4d5a22a78f54ecdc0dc864930f6d3f68c6869f